common/crypto/attestation-api/README.md

```diff
@@ -0,0 +1,91 @@
+# Attestation API
+
+This library implements a single API for generating and verifying hardware-based attestation (e.g., Intel SGX EPID/DCAP attestations).
+The library implementation originally started as part of the Fabric Private Chaincode project, and later spun off and extended.
+
+The library works with Intel SGX EPID attestations, verified through Intel Attestation Service (IAS), and DCAP attestations, verified through Intel Trust Authority (ITA).
+
+## Build
+
+Build the docker image to create an environment with all necessaty dependencies.
+```bash
+cd docker
+make
+```
+
+### Simulation mode
+Start the dev container:
+```bash
+docker run -v <project root>:/project -e SGX_MODE=SIM -it oaa-dev
+```
+
+Simulation mode is the default mode. It does not use actual Intel SGX operations, and tests are conducted on simulated-type attestations.
+
+```bash
+make
+```
+
+### Hardware mode
+Start the dev container, here is an example:
+```bash
+docker run --network host -v <project root>:/project -e SGX_MODE=HW -it oaa-dev
+```
+The container is attached to the host network to download the root certificates.
+Also, enabling the HW mode makes the generated artifacts use actual Intel SGX operations -- and tests are conducted on real attestations.
+
+DCAP support with ITA is built by default, but requires the URL to the ITA CA root certificates (in JWK format). If the URL is not specified, the build will succeed but it won't be able to verify DCAP attestations.
+```bash
+export ITA_ROOT_CACERT_URL=<certificate url>
```

Comment:
You should get that when you subscribe to ITA

Reply:
> You should get that when you subscribe to ITA

How and where do i subscribe? I was expecting some corresponding links here ...

Reply:
Here https://docs.trustauthority.intel.com/main/articles/howto-manage-subscriptions.html
I think the ITA part should probably stay as is since (1) ITA is not free, (2) this is under a hyperledger project.
Rather, we should focus on the DCAP-direct part, which is free and open.

Reply:
> Here https://docs.trustauthority.intel.com/main/articles/howto-manage-subscriptions.html I think the ITA part should probably stay as is since (1) ITA is not free, (2) this is under a hyperledger project. Rather, we should focus on the DCAP-direct part, which is free and open.

Hmm, but as long as we include the code we still should document it and it's use?

Reply:
> > Here https://docs.trustauthority.intel.com/main/articles/howto-manage-subscriptions.html I think the ITA part should probably stay as is since (1) ITA is not free, (2) this is under a hyperledger project. Rather, we should focus on the DCAP-direct part, which is free and open.
> 
> Hmm, but as long as we include the code we still should document it and it's use?

More importantly, above link seems for external folks only?   Maybe i can log in as intel employee directly in ITA portal but that link to very carefully never have a link for the portal!!